paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -358,132 +364,424 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+              to_void_cast<T>(output_data));
+        }
+      } else {
+        auto output_data = output->mutable_data<T>(
+            ctx.GetPlace(), paddle::memory::Allocator::kDefault,
+            handler.GetDstMemorySize());
+        dst_memory_p =
+            handler.AcquireDstMemoryFromPrimitive(to_void_cast<T>(output_data));
+      }
 
-    // create mkldnn memory from input tensors (data/weights)
-    auto user_src_memory_p =
-        handler.AcquireSrcMemory(user_src_md, to_void_cast<T>(input_data));
-    auto user_weights_memory_p = handler.AcquireWeightsMemory(
-        user_weights_md, to_void_cast<T>(filter_data));
+      // create convolution op primitive
+      std::shared_ptr<mkldnn::convolution_forward> conv_p;
+      if (bias) {
+        const T* bias_data = bias->data<T>();
+        auto user_bias_md = platform::MKLDNNMemDesc(
+            {bias_tz}, platform::MKLDNNGetDataType<T>(), memory::format::x);
+        auto user_bias_memory_p =
+            handler.AcquireBiasMemory(user_bias_md, to_void_cast<T>(bias_data));
+
+        auto bias_memory_p = handler.AcquireBiasMemoryFromPrimitive(
+            user_bias_memory_p, pipeline);
+        conv_p = handler.AcquireConvolution(src_memory_p, weights_memory_p,
+                                            bias_memory_p, dst_memory_p);
+      } else {
+        conv_p = handler.AcquireConvolution(src_memory_p, weights_memory_p,
+                                            dst_memory_p);
+      }
 
-    // create reorder primitive if the input format is not the preferred one
-    auto src_memory_p =
-        handler.AcquireSrcMemoryFromPrimitive(user_src_memory_p, pipeline);
-    auto weights_memory_p = handler.AcquireWeightsMemoryFromPrimitive(
-        user_weights_memory_p, pipeline, is_test);
+      // push primitive to stream and wait until it's executed
+      pipeline.push_back(*conv_p);
+      stream(stream::kind::eager).submit(pipeline).wait();
 
-    std::shared_ptr<mkldnn::memory> dst_memory_p;
+      output->set_layout(DataLayout::kMKLDNN);
+      output->set_format(GetMKLDNNFormat(*dst_memory_p));
 
-    if (fuse_residual_conn) {
-      auto residual_param = ctx.Input<Tensor>("ResidualData");
-      auto residual_param_data = residual_param->data<T>();
+    } else {
+      auto* scale_in =
+          ctx.HasInput("Scale_in") ? ctx.Input<Tensor>("Scale_in") : nullptr;
+      auto* scale_in_eltwise = ctx.HasInput("Scale_in_eltwise")
+                                   ? ctx.Input<Tensor>("Scale_in_eltwise")
+                                   : nullptr;
+      auto* scale_weights = ctx.HasInput("Scale_weights")
+                                ? ctx.Input<Tensor>("Scale_weights")
+                                : nullptr;
+      auto* scale_out =
+          ctx.HasInput("Scale_out") ? ctx.Input<Tensor>("Scale_out") : nullptr;
+      bool is_multi_channel = (scale_weights->memory_size() > 1) ? true : false;
+
```

Comment:
Could use Paddle assertions for `scale_in`, `scale_in_eltwise`, `scale_weights` to check whether the variables are nullptr? 

I noticed that these variables are used later in the code without checking if the point to correct (non-null) data.

Reply:
Added